### PR TITLE
add EntryProcessor documentation

### PR DESCRIPTION
**Pull request by furkansenharputlu**
This PR adds documentation for `EntryProcessor`. #340

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1458,6 +1458,91 @@ map.addEntryListener(mapEventListener).then(function () {
...
+EntryProcessorFactory.prototype.create = function (type) {
+    if (type === 1) {
+        return new IdentifiedEntryProcessor();
+    }
+    return null;
+};
+```
+
+Then, You need to implement the Java equivalent of your entry processor and its factory and create your own compiled class or JAR files. For adding your own compiled class or JAR files, please see [Adding User Library to CLASSPATH section](#adding-user-library-to-classpath). Note that you need to configure the `hazelcast.xml` to add your factory. For more details about the Java implementation of the `EntryProcessor`, see [Entry Processor section](https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#entry-processor) in the IMDG Reference Manual.
+
```
Comment (mustafaiman) [inline README.md]: I think the following is also a crucial information for the users:
`The code that runs on entries is implemented in Java on the server side. Client side entry processor is used to specify which entry processor should be called and to configure the entry processor.`

My suggestion is to add this here. Also we need the actual Java implementation of the processor and hazelcast.xml in the sample.

File: README.md [hunk 2]
```diff
@@ -1696,6 +1698,156 @@ map.addEntryListener(mapEventListener).then(function () {
...
+The functions below are in the `IMap` functions for entry processing.
+
+- `executeOnKey` processes an entry mapped by a key.
+
+- `executeOnKeys` processes entries mapped by a list of keys.
+
+- `executeOnEntries` can process all entries in a map with a defined predicate. Predicate is optional.
+
+In the Node.js client, an `EntryProcessor` should be `IdentifiedDataSerializable` or `Portable` because server should be able to deserialize it to process.
```
Comment (Serdaro) [inline README.md]: ... because the server should be ...

File: README.md [hunk 3]
```diff
@@ -1696,6 +1698,156 @@ map.addEntryListener(mapEventListener).then(function () {
...
+
+An `EntryProcessor` which is `IdentifiedDataSerializable` example is shown below.
```
Comment (Serdaro) [inline README.md]: The following is an example for `EntryProcessor` which is `IdentifiedDataSerializable`.

File: README.md [hunk 4]
```diff
@@ -1696,6 +1698,156 @@ map.addEntryListener(mapEventListener).then(function () {
...
+        <data-serializable-factories>
+            <data-serializable-factory factory-id="5">
+                IdentifiedFactory
+            </data-serializable-factory>
+        </data-serializable-factories>
+    </serialization>
+</hazelcast>
+```
+
+The code that runs on entries is implemented in Java on the server side. Client side entry processor is used to specify which entry processor should be called. For more details about the Java implementation of the entry processor, please see [Entry Processor section](https://docs.hazelcast.org/docs/latest/manual/html-single/index.html#entry-processor) in the IMDG Reference Manual.
```
Comment (Serdaro) [inline README.md]: ... runs on the entries ...

Comment (Serdaro) [reply README.md]: .. in the Hazelcast IMDG Reference Manual.

**Pull request merged**